FenwickTree/RangeUpdatePointQueries.java

```diff
@@ -0,0 +1,87 @@
+public class RangeUpdatePointQueries {
+	
+	public static void main(String[] args) {
+		RangeUpdatePointQueries d=new RangeUpdatePointQueries();
+		//input array
+		int[] inArr={0,0,0,0,0,0};
+		//Store the fenwick tree
+		int[] feniTree=new int[inArr.length+1];
+		//Update range using method 2
+		d.updateRangeMeth2(inArr, 2, 4, 10);
+		//Point query method 2
+		System.out.println(d.getPointMeth2(inArr, 3));
+		//Mehth 3 call
+		//Construct fenwick tree 
+		d.contructFeniTree(inArr);
```

Comment:
Why does constructFeniTree not use its return value? Should it not be:
```java
int[] fenwickTree = d.contructFeniTree(inArr);
```

Reply:
I have written both the linear and logarithmic methods operating on same input array. Refactoring will make it clear.